File: icamera/src/main/java/me/shouheng/icamera/enums/FlashMode.java
package me.shouheng.icamera.enums;

import android.support.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import static me.shouheng.icamera.enums.FlashMode.*;

/**
 * Flash mode
 *
 * @author WngShhng (shouheng2015@gmail.com)
 * @version 2019/4/14 21:57
 */
@IntDef({FLASH_ON, FLASH_OFF, FLASH_AUTO})
@Retention(RetentionPolicy.SOURCE)
public @interface FlashMode {

    /** Flash on */
    int FLASH_ON        = 0;

    /** Flash off */
    int FLASH_OFF       = 1;

    /** Auto */
    int FLASH_AUTO      = 2;
}

File: icamera/src/main/java/me/shouheng/icamera/enums/FlashMode.kt
package me.shouheng.icamera.enums

import android.support.annotation.IntDef

/**
 * Flash mode
 *
 * @author WngShhng (shouheng2015@gmail.com)
 * @version 2019/4/14 21:57
 */
@IntDef(FlashMode.FLASH_ON, FlashMode.FLASH_OFF, FlashMode.FLASH_AUTO)
@kotlin.annotation.Retention(AnnotationRetention.SOURCE)
annotation class FlashMode {
    companion object {
        /** Flash on  */
        const val FLASH_ON = 0
        /** Flash off  */
        const val FLASH_OFF = 1
        /** Auto  */
        const val FLASH_AUTO = 2
    }
}